web/client/epics/annotations.js

```diff
@@ -202,23 +202,25 @@ export default (viewer) => ({
         .switchMap(() => {
             const annotationsLayer = annotationsLayerSelector(store.getState());
             if (annotationsLayer) {
+                const {visibility = false, features: annotationFeatures = []} = annotationsLayer;
                 // parsing old style structure
-                let features = (annotationsLayer.features || []).map(ftColl => {
-                    return {...ftColl, style: {}, features: (ftColl.features || []).map(ft => {
-                        let styleType = ft.properties.isCircle && "Circle" || ft.properties.isText && "Text" || ft.geometry.type;
-                        let extraStyles = [];
-                        if (styleType === "Circle") {
-                            extraStyles.push({...DEFAULT_ANNOTATIONS_STYLES.Point, iconAnchor: [0.5, 0.5], type: "Point", title: "Center Style", filtering: false, geometry: "centerPoint"});
-                        }
-                        if (styleType === "LineString") {
-                            extraStyles.concat(getStartEndPointsForLinestring());
-                        }
-                        return {...ft,
-                            style: isArray(ft.style) ? ft.style.map(ftStyle => {
-                                const {symbolUrlCustomized, ...filteredStyle} = ftStyle;
-                                return filteredStyle;
-                            }) : [{...ftColl.style[styleType], id: ftColl.style[styleType].id || uuidv1(), symbolUrlCustomized: undefined}].concat(extraStyles)};
-                    })};
+                let features = annotationFeatures.map(ftColl => {
+                    return {...ftColl,  properties: {...ftColl.properties, visibility}, style: {},
+                        features: (ftColl.features || []).map(ft => {
+                            let styleType = ft.properties.isCircle && "Circle" || ft.properties.isText && "Text" || ft.geometry.type;
+                            let extraStyles = [];
+                            if (styleType === "Circle") {
+                                extraStyles.push({...DEFAULT_ANNOTATIONS_STYLES.Point, iconAnchor: [0.5, 0.5], type: "Point", title: "Center Style", filtering: false, geometry: "centerPoint"});
+                            }
+                            if (styleType === "LineString") {
+                                extraStyles.concat(getStartEndPointsForLinestring());
+                            }
+                            return {...ft,
+                                style: isArray(ft.style) ? ft.style.map(ftStyle => {
+                                    const {symbolUrlCustomized, ...filteredStyle} = ftStyle;
+                                    return filteredStyle;
+                                }) : [{...ftColl.style[styleType], id: ftColl.style[styleType].id || uuidv1(), symbolUrlCustomized: undefined}].concat(extraStyles)};
+                        })};
```

Comment:
Can you describe a little bit with comments what happens here. Is not clear reading the code, expecially the "why". It could be a good imrovement.